src/components/DeckMechanisms/index.js

```diff
@@ -139,18 +138,16 @@ export default class DeckMechanisms extends React.Component {
         // Unless the play is actually free or a discard, decrease the amount
         // of available mana by the cost the card
         if (!(options.free || options.discard)) {
-          newState.mana -= state.deck[deckIds.indexOf(id)].mana
+          newState.mana -= card.mana
         }
 
         return newState
       },
-      () => this.handleCardEffect(id)
+      () => this.handleCardEffect(card)
     )
   }
 
-  handleCardEffect = id => {
-    const card = this.state.deck.find(card => card.id === id)
-
+  handleCardEffect = card => {
```

Comment:
Passing card instead of id to avoid searching for card data again. Not sure if this is better since the argument now contains more data